streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsRebalanceListener.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.processor.internals;
+
+import java.util.Collection;
+import java.util.List;
+import org.apache.kafka.clients.consumer.ConsumerRebalanceListener;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.utils.Time;
+import org.apache.kafka.streams.processor.internals.StreamThread.State;
+import org.apache.kafka.streams.processor.internals.assignment.AssignorError;
+import org.slf4j.Logger;
+
+public class StreamsRebalanceListener implements ConsumerRebalanceListener {
+
+    private final Time time;
+    private final TaskManager taskManager;
+    private final StreamThread streamThread;
+    private final Logger log;
+
+    StreamsRebalanceListener(final Time time,
+        final TaskManager taskManager,
+        final StreamThread streamThread,
+        final Logger log) {
+        this.time = time;
+        this.taskManager = taskManager;
+        this.streamThread = streamThread;
+        this.log = log;
+    }
+
+    @Override
+    public void onPartitionsAssigned(final Collection<TopicPartition> assignedPartitions) {
+        log.debug("at state {}: partitions {} assigned at the end of consumer rebalance.\n" +
+                "\tpreviously assigned active tasks: {}\n" +
+                "\tpreviously assigned standby tasks: {}\n",
+            streamThread.state(),
+            assignedPartitions,
+            taskManager.activeTaskIds(),
```

Comment:
Why we no longer log `current suspended active tasks` any more?

Reply:
I'd say the hardest part about making the listener protocol unaware is figuring out what to log that is relevant and makes sense for both rebalancing protocols :wink: 
My thinking here was, in onPartitions revoked we log which partitions are being revoked and the current assigned tasks in the beginning, then log suspended (active) tasks at the end.
For onPartitionsAssigned:
In cooperative, we may not have called onPartitionsRevoked or logged anything yet at all, so we want to log what the assignment was before the rebalance here. And if onRevoked was not called, nothing was suspended so why bring it up here.
During eager we have already logged the suspended active tasks (in onPartitionsRevoked), plus the "previously assigned active tasks" should be the same as the suspended tasks anyway.
That said, I don't feel too strongly about it so we can log that in addition to the above even if it may be empty in cooperative

Reply:
I think I'm convinced :) Let's keep them as removed.